elasticsearch/client.go

```diff
@@ -71,7 +82,9 @@ func (cli *Client) post(ctx context.Context, index string, docType string, actio
 	}
 
 	if cli.signRequests {
-		awsauth.Sign(req)
+		if err = cli.awsSDKSigner.Sign(req, reader, time.Now()); err != nil {
+			return nil, err
+		}
```

Comment:
I am being picky as it shouldn't ever happen but I think  you should check that `awsSDKSigner` is not `nil` if `signRequests == true`. 

With the current code that shouldn't be possible _but_ if this was every refactored its possible that assumption could be missed and a `nil` value could slip through and cause pain. If we add the check we can give an explicit error message to state whats wrong.

Reply:
This is being handled by the dp-elasticsearch library: https://github.com/ONSdigital/dp-elasticsearch/blob/main/awsauth/sign.go#L55. I've added a test to check that the correct error message is being returned.